task2/js/script.js

```diff
@@ -0,0 +1,125 @@
+function isArray(something) {
+  return something instanceof Array;
+}
+
+function isArray2(something) {
+  return Object.prototype.toString.call(something) === '[object Array]';
+}
+
+function range(from, to, step) {
+  var retArray = [];
+  var i;
+
+  if (from == undefined)
+    return retArray;
+
+  if (step == undefined)
+    step = 1;
+
+  if (to == undefined) {
+    to = from;
+    from = 0;
+  }
+
+  for (i = from; i < to; i += step)
+    retArray.push(i);
+
+  return retArray;
+}
+
+function compact(someArray) {
+  var retArray = someArray.map(function () {
+    return true;
+  });
+
+  return retArray;
+}
+
+function compact2(someArray) {
+  var retArray = [];
+  var i;
+
+  for (i = 0; i < someArray.length; i++)
+    retArray.push(true);
+
+  return retArray;
+}
+
+function sum(someArray) {
+  var returnResult = someArray.reduce(function (sum, current) {
+    return sum + current;
+  }, 0);
+
+  return returnResult;
+}
+
+function sum2(someArray) {
+  var returnResult = 0;
+  var i;
+
+  for (i = 0; i < someArray.length; i++)
+    returnResult += someArray[i];
+
+  return returnResult;
+}
+
+function unique(someArray) {
```

Comment:
in doesn't work with objects, functions etc
I explained that toString is called to make an object key